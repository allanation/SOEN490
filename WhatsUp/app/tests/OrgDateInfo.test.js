import "react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from "@testing-library/react-native";
import OrgPOCScreen from "../screens/OrgPointOfContact";
import OrgDateInfoScreen from "../screens/OrgDateInfo";
import IOSDateTimePicker from "../components/IOSDateTimePicker";
import OrgDayScheduleScreen from "../screens/OrgDaySchedule";
import { Alert } from "react-native";

jest.useFakeTimers();

// it("Renders Create New Event Third Page Correctly", () => {
//   const tree = render(
//     <NavigationContainer>
//       <OrgDateInfoScreen />
//     </NavigationContainer>
//   ).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test("If the dates are missing, an alert should be prompted", async () => {
  render(
    <NavigationContainer>
      <OrgDateInfoScreen />
    </NavigationContainer>
  );

  jest.spyOn(Alert, "alert");

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
    expect(Alert.alert);
  });
});

test("Successfully go to next page when the start date and end date are filled", async () => {
  render(
    <NavigationContainer>
      <IOSDateTimePicker />
    </NavigationContainer>
  );

  const startPicker = await screen.getByTestId('startTime');
  fireEvent(startPicker, 'onChange', { nativeEvent: { timestamp: new Date('2023-03-04') } });

  const endPicker = await screen.getByTestId('endTime');
  fireEvent(endPicker, 'onChange', { nativeEvent: { timestamp: new Date('2023-04-04') } });

  render(
    <NavigationContainer>
      <OrgDateInfoScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("nextButton"));
  });
});

test("Successfully go back to second create event page when clicking on go back Icon", async () => {
  render(
    <NavigationContainer>
      <OrgDateInfoScreen />
    </NavigationContainer>
  );

  await waitFor(() => {
    fireEvent.press(screen.getByTestId("backButton"));
  });
  render(
    <NavigationContainer>
      <OrgPOCScreen />
    </NavigationContainer>
  );
  expect(screen.getByPlaceholderText("Name")).toBeTruthy();
});