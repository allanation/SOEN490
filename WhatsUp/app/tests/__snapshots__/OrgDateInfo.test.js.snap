// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Create New Event Third Page Correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "marginTop": 30,
        "padding": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "elevation": 2,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "marginLeft": 2,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontSize": 28,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Set Date Information
    </Text>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
            "marginLeft": "1%",
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "#747474",
            "fontSize": 20,
            "marginBottom": 6,
            "marginLeft": "1%",
            "paddingVertical": 6,
          },
        ]
      }
    >
      Please pick the dates for your event
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingTop": 15,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#989797",
            "borderRadius": 12,
            "borderWidth": 1,
            "marginLeft": "1%",
            "marginVertical": 10,
            "opacity": 0.9,
            "paddingHorizontal": 12,
            "paddingVertical": 7,
            "width": "98%",
          },
          Object {
            "elevation": 2,
            "justifyContent": "center",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 3,
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Number of days"
        placeholderTextColor="grey"
        style={
          Object {
            "color": "black",
            "fontFamily": "Helvetica Neue",
            "fontSize": 18,
            "opacity": 1,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "paddingTop": 2,
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "color": "#484545",
              "flex": 1,
              "flexDirection": "row",
              "fontSize": 24,
              "marginTop": 10,
              "paddingTop": 50,
            }
          }
        >
           Start Date & Time 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "flexDirectionL": "row",
              "justifyContent": "center",
              "padding": 20,
            }
          }
        >
          <RNDateTimePicker
            date={1680372507598}
            displayIOS="inline"
            enabled={true}
            minuteInterval="15"
            mode="time"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 49.5,
                },
                Object {
                  "accentColor": "red",
                  "marginBottom": 10,
                  "width": "40%",
                },
              ]
            }
            testID="startTime"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "lightgrey",
              "flex": 1,
              "height": 1,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "color": "#484545",
              "flex": 1,
              "flexDirection": "row",
              "fontSize": 24,
              "marginTop": 10,
              "paddingTop": 50,
            }
          }
        >
           End Date & Time 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "flexDirectionL": "row",
              "justifyContent": "center",
              "padding": 20,
            }
          }
        >
          <RNDateTimePicker
            date={1680372507599}
            displayIOS="inline"
            enabled={true}
            minuteInterval="15"
            mode="time"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 49.5,
                },
                Object {
                  "accentColor": "red",
                  "marginBottom": 10,
                  "width": "40%",
                },
              ]
            }
            testID="endTime"
          />
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#00c0a4",
              "borderRadius": 20,
              "elevation": 4,
              "marginLeft": "1%",
              "marginVertical": 40,
              "opacity": 1,
              "padding": 12,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
          >
            Confirm Dates
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#989797",
        }
      }
    >
      You must confirm your event dates before proceeding.
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#00c0a4",
          "borderRadius": 20,
          "elevation": 4,
          "marginLeft": "1%",
          "marginVertical": 40,
          "opacity": 1,
          "padding": 12,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
          "width": "90%",
        }
      }
      testID="nextButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Helvetica Neue",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            undefined,
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</View>
`;
